junto-rust/zomes/core/code/src/expressions/definitions/channel_entry_definitions.rs

```diff
@@ -0,0 +1,173 @@
+//Entry Definition(s)
+use super::app_definitions;
+
+use hdk::{
+    entry_definition::ValidatingEntryType,
+    holochain_core_types::{
+        dna::entry_types::Sharing,
+        cas::content::Address
+    }
+};
+
+pub fn channel_definition() -> ValidatingEntryType {
+    entry!(
+        name: "channel",
+        description: "Channel Object Entry",
+        sharing: Sharing::Public,
+        //native_type: app_definitions::Channel,
+        validation_package: || {
+            hdk::ValidationPackageDefinition::Entry
+        },
+
+        validation: |validation_data: hdk::EntryValidationData<app_definitions::Channel>| {
+            Ok(())
+        },
+
+        links: [
```

Comment:
This all looks like a repetition of the relationships in `app_definition.rs`. You could write a macro that creates these from! definitions from those. Or even just map over the link definitions to create a vector of `hdk::entry_definition::ValidatingLinkDefinition` which is all the entry! macro is expecting.